### ISSUE TITLE
Request: Better documentation for the chart API, starting with the legend

### ISSUE THREAD

**Issue by szalapski** [novus/nvd3#1950]
Would anyone be willing to update the documentation to provide more information on how to programmatically interact with a chart?    I see there is an object chart.legend which seems to have the [most basic documentation here](https://github.com/novus/nvd3/wiki/nv.models.legend).  

So, for example, what objects are on chart.legend and how do I use them?  If I want to get the "disabled" state of each series on the chart, where is that array?  If I want the function that is pre-bound to a state change of a bullet in the legend, where is that?  

My immediate need is to save the "disabled" state of the bullets to persist for the user, but I will likely have other needs in the future as well around the legend API.

**Comment by liquidpele** [comment]
https://nvd3-community.github.io/nvd3/examples/documentation.html

https://github.com/novus/nvd3/wiki/Data-Attributes

**Comment by szalapski** [comment]
Great, so I'd love to see the content at https://nvd3-community.github.io/nvd3/examples/documentation.html fleshed out more.  What is legend.updateState?  Can legend.dispatch  be added?  How do I use legend.dispatch?  How do I get the function that is prebound so that I can called from my new function I pass in?

**Comment by liquidpele** [comment]
I created that doc page to actually build the structure based off the nvd3 running code, so the options given there are always up to date in that respect... to fill out the details for each one, see the big json dictionary at the bottom of the documention.html file.  

When I created that, I was essentially reverse engineering things so some things might be empty still or slightly wrong, so feel free to fill in any gaps etc and send me pull requests :)

**Comment by szalapski** [comment]
Can you help with any of my particular questions above?

**Comment by liquidpele** [comment]
Not really... you'll want to view the examples in the examples directory, and read the source as needed.   dispatch is basic d3.js dispatching.  This is an open source project and I'm not the original author or anything.  If you require detailed support, I'd recommend amcharts or highcharts.   Cheers.

**Comment by jeznag** [comment]
@szalapski ` If I want to get the "disabled" state of each series on the chart, where is that array?`
chart.datum()

Like liquidpele said, reading the source is the best way to understand what the code does.

**Comment by d-damien** [comment]
Same as @szalapski + how to do the opposite (update it from the outside).
